controllers/space/space_controller_test.go

```diff
@@ -737,16 +737,126 @@ func TestPromoteSpace(t *testing.T) {
 		})
 	})
 
+	t.Run("tier update (same tier but updated references)", func(t *testing.T) {
+		// given that Space is set to the same tier that has been updated and the corresponding NSTemplateSet is not up-to-date
+		s := spacetest.NewSpace("oddity",
+			spacetest.WithTierNameAndHashLabelFor(basicTier), // assume that at this point, the `TemplateTierHash` label was already removed by the ChangeTierRequestController
+			spacetest.WithSpecTargetCluster("member-1"),
+			spacetest.WithStatusTargetCluster("member-1"), // already provisioned on a target cluster
+			spacetest.WithFinalizer())
+		hostClient := test.NewFakeClient(t, s, basicTier, otherTier)
+		nstmplSet := nstemplatetsettest.NewNSTemplateSet("oddity", nstemplatetsettest.WithReadyCondition()) // NSTemplateSet has references to old basic tier
+		member1Client := test.NewFakeClient(t, nstmplSet)
+		member1 := NewMemberClusterWithClient(member1Client, "member-1", corev1.ConditionTrue)
+		member2 := NewMemberCluster(t, "member-2", corev1.ConditionTrue)
+		ctrl := newReconciler(hostClient, member1, member2)
+
+		// when
+		res, err := ctrl.Reconcile(context.TODO(), requestFor(s))
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: 1 * time.Second,
+		}, res) // explicitly requeue while the NSTemplate update is triggered by its controller
+		spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
+			Exists().
+			HasTier(basicTier.Name).
+			HasSpecTargetCluster("member-1").
+			HasStatusTargetCluster("member-1").
+			HasConditions(spacetest.Updating()).
+			HasLabel(tierutil.TemplateTierHashLabelKey(basicTier.Name))
+		nsTmplSet := nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, "oddity", member1.Client).
+			Exists().
+			HasTierName(basicTier.Name).
+			Get()
+		require.False(t, tierutil.TierHashMatches(basicTier, nstmplSet.Spec))
```

Comment:
This should match! The NSTemplateSet.spec should be updated. 
```suggestion
		require.True(t, tierutil.TierHashMatches(basicTier, nsTmplSet.Spec))
```

Reply:
😱 Thank you for noticing this! Fixed in 12fab16b5184a9025ff9901c49ff62d0fd4004d1 I didn't notice that I had `nstmplSet` and `nsTmplSet` vars so I fixed that too.